strapi 4.14.0: no npm run develop compatibility with v.2.3.4 of strapi-plugin-slugify

## System configuration

- Strapi Version: 4.14.0
- Operating System: MacOS 12.7
- Database: mySQL 8.0.34 (using mysql2 as driver)
- Node Version: 18.12.1
- NPM Version: 9.8.0
- Yarn Version: none
- Dev language: typescript

## Describe the problem
Installing strapi v.4.14.0,  the `npm run develop` command fail with a blocking error message in the log.
Uninstalling `strapi-plugin-slugify` plugin, strapi works fine again.

### Error log
~~~code
Error: Could not load js config file .../server/node_modules/strapi-plugin-slugify/strapi-server.js: Cannot find module '@strapi/strapi/lib/core-api/controller/transform'
Require stack:
- .../server/node_modules/strapi-plugin-slugify/server/controllers/slug-controller.js
- .../server/node_modules/strapi-plugin-slugify/server/controllers/index.js
- .../server/node_modules/strapi-plugin-slugify/server/index.js
- .../server/node_modules/strapi-plugin-slugify/strapi-server.js
- .../server/node_modules/@strapi/utils/dist/import-default.js
- .../server/node_modules/@strapi/utils/dist/index.js
- .../server/node_modules/@strapi/data-transfer/dist/utils/components.js
- .../server/node_modules/@strapi/data-transfer/dist/strapi/queries/entity.js
- .../server/node_modules/@strapi/data-transfer/dist/strapi/queries/index.js
- .../server/node_modules/@strapi/data-transfer/dist/strapi/providers/local-destination/strategies/restore/index.js
- .../server/node_modules/@strapi/data-transfer/dist/strapi/providers/local-destination/strategies/index.js
- .../server/node_modules/@strapi/data-transfer/dist/strapi/providers/local-destination/index.js
- .../server/node_modules/@strapi/data-transfer/dist/strapi/providers/index.js
- .../server/node_modules/@strapi/data-transfer/dist/strapi/index.js
- .../server/node_modules/@strapi/data-transfer/dist/index.js
- .../server/node_modules/@strapi/strapi/dist/commands/index.js
- .../server/node_modules/@strapi/strapi/bin/strapi.js
    at loadJsFile (.../server/node_modules/@strapi/strapi/dist/core/app-configuration/load-config-file.js:21:19)
    at loadFile (.../server/node_modules/@strapi/strapi/dist/core/app-configuration/load-config-file.js:41:20)
    at Object.loadPlugins (.../server/node_modules/@strapi/strapi/dist/core/loaders/plugins/index.js:93:62)
    at async Strapi.loadPlugins (.../server/node_modules/@strapi/strapi/dist/Strapi.js:373:9)
    at async Promise.all (index 3)
    at async Strapi.register (.../server/node_modules/@strapi/strapi/dist/Strapi.js:407:9)
    at async Strapi.load (.../server/node_modules/@strapi/strapi/dist/Strapi.js:493:9)
    at async workerProcess (.../server/node_modules/@strapi/strapi/dist/commands/actions/develop/action.js:100:28)
    at async exports.default (.../server/node_modules/@strapi/strapi/dist/commands/actions/develop/action.js:38:20)
~~~

## Steps to reproduce the behavior
1. change package.json with the new version of strapi
2. invoke `npm install` command
3. invoke `npm run build` command
4. invoke `npm run develop` command

## Additional context

### package.json
~~~code
{
  "name": "server",
  "private": true,
  "version": "1.0.0",
  "description": "description",
  "scripts": {
    "develop": "strapi develop",
    "start": "strapi start",
    "build": "strapi build",
    "strapi": "strapi"
  },
  "dependencies": {
    "@strapi/plugin-i18n": "4.14.0",
    "@strapi/plugin-users-permissions": "4.14.0",
    "@strapi/provider-email-nodemailer": "^4.14.0",
    "@strapi/strapi": "4.14.0",
    "mysql2": "^3.6.1",
    "strapi-plugin-duplicate-button": "^1.1.13",
    "strapi-plugin-slugify": "^2.3.4"
  },
  "author": {
    "name": "a name"
  },
  "strapi": {
    "uuid": "xxx"
  },
  "engines": {
    "node": ">=14.19.1 <=18.x.x",
    "npm": ">=6.0.0"
  },
  "license": "license"
}

~~~

### config/plugin.ts

~~~js
//...

slugify: {
        enabled: true,
        config: {
            shouldUpdateSlug: true,
            contentTypes: {
                teacher: {
                    field: 'slug',
                    references: ['firstname', 'lastname'],
                },
                student: {
                    field: 'slug',
                    references: ['firstname', 'lastname', 'birthdate'],
                },
            },
        },
      },

//...
~~~

I had the same problem here.

the problem is in the file `slug-controller`: Cannot find module '@strapi/strapi/lib/core-api/controller/transform. (line 6).

--- a/server/controllers/slug-controller.js
+++ b/server/controllers/slug-controller.js
@@ -3,7 +3,7 @@
 const _ = require('lodash');
 const { NotFoundError } = require('@strapi/utils').errors;
 const { getPluginService } = require('../utils/getPluginService');
-const { transformResponse } = require('@strapi/strapi/lib/core-api/controller/transform');
+const { transformResponse } = require('@strapi/strapi/dist/core-api/controller/transform');
 const { isValidFindSlugParams } = require('../utils/isValidFindSlugParams');
 const { sanitizeOutput } = require('../utils/sanitizeOutput');
 const { hasRequiredModelScopes } = require('../utils/hasRequiredModelScopes');

